File: backend/src/main/java/com/eudext/erp/documents/internal/attachment/ScanStatus.java
package com.eudext.erp.documents.internal.attachment;

/** DOC-4: outcome of the virus scan run on an attachment at upload time. */
public enum ScanStatus {
    /** No scanner configured in this environment (see {@code eudext.documents.attachments.virus-scan.enabled}). */
    PENDING,
    CLEAN,
    INFECTED,
    FAILED
}

File: frontend/src/lib/documents/scan-status.ts
/** Mirrors the backend's `ScanStatus` enum (DOC-4). */
export const SCAN_STATUSES = ["PENDING", "CLEAN", "INFECTED", "FAILED"] as const;

export type ScanStatus = (typeof SCAN_STATUSES)[number];

export const SCAN_STATUS_LABELS: Record<ScanStatus, string> = {
  PENDING: "Scan pending",
  CLEAN: "Clean",
  INFECTED: "Infected",
  FAILED: "Scan failed",
};
